File: test/T3L90_ok.pas
(* Test file for level 100 of CSCE 531 Pascal compiler Project III *)

(* There are no errors in this file *)

program T3L100ok;

var
   a,b : Integer;
   c   : Char;


procedure print_one; External;
procedure print_two; External;
procedure print_three; External;
procedure print_dunno; External;
procedure print_just; External;
procedure print_isnt; External;
procedure print_hello; External;
procedure print_oops; External;
procedure print_712; External;
procedure print_is_the; External;
procedure print_grooviest; External;
procedure print_newline; External;


begin

   a := 9;
   b := 3;

   case a of
     1 : print_one;
     2 : print_two;
     3 : print_three
   end; { case }

   case a of
     1 : print_one;
     2 : print_two;
     3 : print_three
   else
      print_dunno
   end; { case }

   a := 2;

   case a of
     1 : print_one;
     2 : print_two;
     3 : print_three
   else
      print_dunno
   end; { case }

   a := 1;

   case a of
     1,2 : print_one;
     3   : print_three
   else
      print_dunno
   end; { case }

   a := 2;

   case a of
     1,2 : print_one;
     3   : print_three
   else
      print_dunno
   end; { case }

   a := 3;

   case a of
     1,2 : print_one;
     3   : print_three
   else
      print_dunno
   end; { case }


   { This next part is a lot easier with constant-folding. }

   c := chr(3);

   case c of
     chr(3) :
        begin
	   case c of
	     chr(1),chr(2),chr(4) : print_oops;
	     chr(3)		  : 
	        begin
		   print_712;
		   print_is_the
		end
	   end; { case }
	   print_grooviest
	end
   else	       
      print_oops;
      case ord(c) of
	1,2,3,4	: print_oops
      else	
	 print_oops
      end; { case }
   end; { case }
   
   print_newline

end.

File: Part3/T3L90_ok.c
long int A,B;
unsigned char C;


void Print_one();
void Print_two();
void Print_three();
void Print_dunno();
void Print_just();
void Print_isnt();
void Print_hello();
void Print_oops();
void Print_712();
void Print_is_the();
void Print_grooviest();
void Print_newline();


int main()
{
    A = 9;
    B = 3;

    switch (A) {
        case 1 : Print_one(); break;
        case 2 : Print_two(); break;
        case 3 : Print_three();
    }

    switch (A) {
        case 1 : Print_one(); break;
        case 2 : Print_two(); break;
        case 3 : Print_three(); break;
        default :
            Print_dunno();
    }

    A = 2;

    switch (A) {
        case 1 : Print_one(); break;
        case 2 : Print_two(); break;
        case 3 : Print_three(); break;
        default :
            Print_dunno();
    }

    A = 1;

    switch (A) {
        case 1 :
        case 2 : Print_one(); break;
        case 3 : Print_three(); break;
        default :
            Print_dunno();
    }

    A = 2;

    switch (A) {
        case 1 :
        case 2 : Print_one(); break;
        case 3   : Print_three(); break;
        default :
            Print_dunno();
    }

    A = 3;

    switch (A) {
        case 1 :
        case 2 : Print_one(); break;
        case 3 : Print_three(); break;
        default :
            Print_dunno();
    }


        /* This next part is a lot easier with constant-folding. */

    C = 3;

    switch (C) {
        case 3 :
        
            switch (C) {
                case 1 :
                case 2 :
                case 4 : Print_oops(); break;
                case 3 : 
	        
                    Print_712();
                    Print_is_the();
            }
            Print_grooviest(); break;
	
        default :	       
            Print_oops();
            switch (C) {
                case 1 :
                case 2 :
                case 3 :
                case 4 : Print_oops(); break;
                default :	
                    Print_oops();
            }
    }
   
    Print_newline();

    return 0;
}

